resources/assets/js/components/H2Components/utils.ts

```diff
@@ -0,0 +1,13 @@
+export interface GeneralProps {
+  /** The standard css class attribute */
+  className?: string;
+}
+
+export interface GeneralBtnProps {
+  /** The hydrogen styling for a button component */
+  buttonStyling?: string;
+  /** The type of button: "submit" | "reset" | "button" */
+  type?: "submit" | "reset" | "button";
+  /** The onClick event handler */
+  onClick?: () => void;
```

Comment:
To be fully generic, the onClick definition should be `(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void`. Callback functions that we actually pass in can ignore the event parameter.